Update PR template (#3290)
canonical-web-and-design/snapcraft.io: .github/pull_request_template.md

## Done

> - _A good changelist description explains what the change achieves, at a high level, and why you’re making this change_

Reviewer: ```suggestion
- _A changelist description explaining what the change achieves and why you’re making it._
```

> ### Done

Reviewer 2: I prefer the h2 styling but :shrug: saves adding artificial hr's

> 

Reviewer 2: This might just be a personal gripe but I remove the newlines under titles every time. I prefer to have headings and content next to each other and rely on the typography to space them nicely.

> - [Add clear and detailed changes description]

Reviewer 2: ```suggestion
- _A good changelist description explains what the change achieves, at a high level, and why you’re making this change._
```

> ---

Reviewer: It's nice looking but I think we should keep the description as minimal as possible. The titles are already a good way to show a text septation.

### How to QA

> - Visit the site in your browser at: [Replace with demo link]

Reviewer 2: Let's save ourselves time and effort by just saying "Visit the demo in your browser, link commented below."

> - Visit the site in your browser at: [Replace with demo link]

Reviewer: I don't think we should remove completely the dotrun commands references and local development references. 
Demos are only available for people in the organisation. We should find a way to keep what we had but also suggest to add demo link in the PR description for us.

Author: There will be a link to https://github.com/canonical-web-and-design/snapcraft.io/blob/master/README.md#local-development where you can find all the info about the local development. Do you think that's not enough?

Reviewer: well when reviewing a PR I don't want to be clicking on some random link that sends me away of the PR. I strongly think that having this link already there is important. 

The strength of dotrun is that it a unique command to run. I believe that having it with the link in the PR template is not noise but it is actually helpful.

I think the draft exercise is good thing but not a pattern and we should keep a local first QA and let the developer then replace the links with demo if she wants.

Author: updated

- To run the project locally, please refer to the [Readme](https://github.com/canonical-web-and-design/snapcraft.io/blob/master/README.md#local-development) file for more information

> - If your demo doesn't work, read more about the [demoservice](https://discourse.ubuntu.com/t/demoservice/16862)

Reviewer 2: What about moving this as help (smaller) text under the first bullet point about visiting the demo?
